### PR TITLE
op-chain-ops: Forge scripts improvements

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

This PR isolates the forge-scripts improvements from the interop devnet work.

Changes:
- pull in updated op-geth, to pull in the https://github.com/ethereum-optimism/op-geth/pull/366 change
- Add `WithPrecompileAtAddress` to easily attach (and cleanup) precompiles
- Add `WithScript` to easily attach (and cleanup) script contracts with bindings
- Add `ResetNonce` cheatcode
- Fix unintended contract bytecode byte-slice sharing
- Fix the `prank` cheatcode:
  - The caller-override in the previous approach was not being applied early enough, causing contract-creation to result in a different address than expected. With the new geth EVM caller-override hook this is now much more trivial to get right.
  - Fix the `startBroacast` cheatcode to properly apply the caller prank
  - Isolate the prank code from the script-host code
- Isolate state-dump cheatcode functionality as helper method, and improve it by cleaning up known script accounts.
- Use the improved source-map utils for better debugging
- Make precompile-overrides all use the same precompile-map, rather than having special cases for the cheatcode and console precompile.
- Option on host to disable the EVM bytecode-size check, to run larger scripts.
- A few script-host convenience methods, to do common things like calls, wiping an account, etc.

Depends on https://github.com/ethereum-optimism/optimism/pull/11574

**Tests**

Existing script test covers the basic script functionality, other more finer forge-script behavior is covered through the deployment-test (coming with interop-devnet PR).

**Metadata**

Separated out of https://github.com/ethereum-optimism/optimism/pull/11324

### CODE REVIEW

File: cannon/testdata/example/alloc/go.mod [hunk 1]
```diff
@@ -7,8 +7,8 @@ toolchain go1.21.1
 require github.com/ethereum-optimism/optimism v0.0.0
 
 require (
-	golang.org/x/crypto v0.26.0 // indirect
```
Comment (tynes) [inline cannon/testdata/example/alloc/go.mod]: This seems like unnecessary diff?

**Review by tynes — APPROVED** [review]

File: cannon/testdata/example/alloc/go.mod (hunk 1 above)
Comment (protolambda) [reply cannon/testdata/example/alloc/go.mod]: It is forced because of the op-geth update

Comment (protolambda) [reply cannon/testdata/example/alloc/go.mod]: Geth rc.4 landed on monorepo develop branch, this diff is now dropped

**Comment by protolambda** [comment]
Rebased on `develop`
